packages/react/src/components/ContainedList/ContainedList.js

```diff
@@ -33,17 +33,39 @@ function ContainedList({
     className
   );
 
+  const filteredChildren = children?.filter(
+    (child) =>
+      !['Search', 'ExpandableSearch'].includes(child?.type?.displayName)
+  );
+
+  const isActionSearch = ['Search', 'ExpandableSearch'].includes(
+    action?.type?.displayName
+  );
```

Comment:
The only caveat I see is that this is locked to components specifically named `Search` and `ExpandableSearch`. If a user wants to use their own component similar to `Search` but named something else, the component won't get any of the styling or rendering help we've provided. The children won't be filtered, etc.

I kind of wish we would've kept ContainedList experimental a little bit longer 😅 

If we had the ability to make a breaking change, I'd suggest making the header its own component, ContainedListHeader. This way we'd have options to do things like
```jsx
<ContainedList>
    <ContainedListHeader action={<ExpandableSearch/>}/>
    <ContainedListItem/>
    <ContainedListItem/>
</ContainedList>
```

or

```jsx
<ContainedList>
    <ContainedListHeader>
        <Search/>
    </ContainedListHeader>
    <ContainedListItem/>
    <ContainedListItem/>
</ContainedList>
```

`ContainedListHeader` could then always wrap any provided children with the class used for styling the sticky header, enabling more flexibility without having to snoop children `displayName`s. Also from my testing, the persistent Search doesn't appear to need to be rendered within the `ul`, it works as a sibling of `.${prefix}--contained-list__header`

---

Could we achieve this without a breaking change via conditional rendering?

* Make the new ContainedListHeader component
* Deprecate props previously used in the header section of ContainedList, point users to ContainedListHeader
* Leave the existing header code in ContainedList, but if ContainedList is given a child of `ContainedListHeader`, don't render the existing code. Instead render the ContainedListHeader. If it's not given a ContainedListHeader, render the existing code.
* Update documentation and stories to use ContainedListHeader
* In v12, remove the existing header code and condition looking for ContainedListHeader as a child

Reply:
@tay1orjones Great feedback - thank you :)   @francinelucca 's comment was the same thought I had, curious your thoughts!